### DISCUSSION TITLE
Get post data in custom validation

### DISCUSSION THREAD

**Discussion by brightauto** [typestack/class-validator#1959] [Questions & Answers]
Hi,

I have used nestjs microservice.
I used class-validator to validate the entire project.
I need to access the all post params inside the custom decorator validator inside nested validator, But i couldn't get this.
How to get all post params?
Kindly give me any suggesstion.

**Comment by sunnypatell** [comment]
@brightauto in a custom `ValidatorConstraintInterface`, the `validate` method receives `ValidationArguments` as the second parameter. use `args.object` to access the entire DTO being validated:

```ts
@ValidatorConstraint({ async: false })
export class MyCustomValidator implements ValidatorConstraintInterface {
  validate(value: any, args: ValidationArguments) {
    const fullBody = args.object as MyDto;
    return fullBody.someOtherField !== value;
  }

  defaultMessage(args: ValidationArguments) {
    return 'Validation failed';
  }
}
```

`args.object` gives you the immediate parent object containing the field being validated, so for nested validators it refers to the parent, not the root. for root-level DTOs, it's the full body.

other useful properties on `ValidationArguments`: `args.property` (field name), `args.targetName` (class name), `args.constraints` (decorator params).

ref: [class-validator custom validation](https://github.com/typestack/class-validator#custom-validation-classes)
